### ISSUE TITLE
Setting sampler state filter on default Linear sampler

### ISSUE THREAD

**Issue by totallyeviljake** [mono/cocos2d-xna#71]
CCrenderTexture:

```
        m_pTexture = new CCTexture2D();
        m_pTexture.SetAliasTexParameters();
```

CCtexture2D:
        public void SetAliasTexParameters()
        {
            m_samplerState.Filter = TextureFilter.Linear;
        }

Crash with an exception because the CCTexture2D sampler state is using the SamplerState.Linear. This sampler can not be changed after it has been bound once, has happened once the extensions test has loaded for the first time. This error happens in the second extensions test.

**Comment by kjpou1** [comment]
Jake

From the Extensions menu option which is the penultimate option right?  

I don't get a crash with this.

**Comment by totallyeviljake** [comment]
you're on MonoGame, right? I wonder if MG implemented SamplerState as a struct instead of class.

**Comment by kjpou1** [comment]
Yes on MG.  I was thinking you were on XNA.  It kind of makes sense and should have remembered that about XNA.  I ran into it on XNA when porting a sample.  It just works differently.  I started to implement the functionality a long time ago in MG but did not understand the SamplerState very well back then.  Never got back to it and am sure it is really not a high priority on the MG Team's list.  I think Steve and Tom did a great job in that implementation.  A heck of a lot better than myself fumbling around with it.

It should be fixed now though in the PR that you merged.
